packages/client/test/rpc/eth/getBlockByNumber.spec.ts

```diff
@@ -68,142 +67,122 @@ const method = 'eth_getBlockByNumber'
-    await baseRequest(server, req, 200, expectRes)
+    const res = await rpc.request(method, [10, true])
+    assert.equal(res.error.code, INVALID_PARAMS)
+    assert.ok(res.error.message.includes('invalid argument 0: argument must be a string'))
   })
 
   it('call with invalid block number', async () => {
     const manager = createManager(createClient({ chain: createChain() }))
-    const server = startRPC(manager.getMethods())
+    const rpc = getRpcClient(startRPC(manager.getMethods()))
 
-    const req = params(method, ['WRONG BLOCK NUMBER', true])
-    const expectRes = checkError(
-      INVALID_PARAMS,
-      'invalid argument 0: block option must be a valid 0x-prefixed block hash or hex integer, or "latest", "earliest" or "pending"'
+    const res = await rpc.request(method, ['WRONG BLOCK NUMBER', true])
+    assert.equal(res.error.code, INVALID_PARAMS)
+    assert.ok(
+      res.error.message.includes(
+        'invalid argument 0: block option must be a valid 0x-prefixed block hash or hex integer, or "latest", "earliest" or "pending"'
+      )
     )
-
-    await baseRequest(server, req, 200, expectRes)
   })
 
   it('call without second parameter', async () => {
     const manager = createManager(createClient({ chain: createChain() }))
-    const server = startRPC(manager.getMethods())
+    const rpc = getRpcClient(startRPC(manager.getMethods()))
 
-    const req = params(method, ['0x0'])
-    const expectRes = checkError(INVALID_PARAMS, 'missing value for required argument 1')
-    await baseRequest(server, req, 200, expectRes)
+    const res = await rpc.request(method, ['0x0'])
+    assert.equal(res.error.code, INVALID_PARAMS)
+    assert.ok(res.error.message.includes('missing value for required argument 1'))
   })
 
   it('call with invalid second parameter', async () => {
     const manager = createManager(createClient({ chain: createChain() }))
-    const server = startRPC(manager.getMethods())
+    const rpc = getRpcClient(startRPC(manager.getMethods()))
 
-    const req = params(method, ['0x0', 'INVALID PARAMETER'])
-    const expectRes = checkError(INVALID_PARAMS)
-    await baseRequest(server, req, 200, expectRes)
+    const res = await rpc.request(method, ['0x0', 'INVALID PARAMETER'])
+    assert.equal(res.error.code, INVALID_PARAMS)
   })
 
   it('call with transaction objects', async () => {
     const manager = createManager(createClient({ chain: createChain() }))
-    const server = startRPC(manager.getMethods())
-    const req = params(method, ['latest', true])
+    const rpc = getRpcClient(startRPC(manager.getMethods()))
+    const res = await rpc.request(method, ['latest', true])
 
-    const expectRes = (res: any) => {
-      assert.equal(typeof res.body.result.transactions[0], 'object', 'should include tx objects')
-    }
-    await baseRequest(server, req, 200, expectRes)
+    assert.equal(typeof res.result.transactions[0], 'object', 'should include tx objects')
```

Comment:
Could we check for something else (e.g. a property, or length of transactions) instead of typeof object? One gotcha with a `typeof myVar === 'object'` is that `typeof null === 'object'` equals true .